### PR TITLE
Convert from gradle to Maven

### PR DESCRIPTION
**Pull request by frenchfrywpepper**

### CODE REVIEW

**Comment by malafeev** [comment]
it would be great to have an issue explaining reasons to migrate to maven

**Comment by frenchfrywpepper** [comment]
@malafeev There are several reasons, but the primary is to enable us to provide better support for our shaded jar. An example issue with the shaded jar is #106 

Another reason is that the Gradle submodule for the Benchmark code is not recognized in IntelliJ. Small nuisances like this are another reason.

File: examples/src/main/java/com/lightstep/tracer/jre/example/Simple.java [hunk 1]
```diff
@@ -9,19 +16,13 @@
...
-import io.opentracing.Span;
-import io.opentracing.SpanContext;
-import io.opentracing.Tracer;
-import io.opentracing.propagation.Format;
-import io.opentracing.propagation.TextMap;
-
 public class Simple {
     public static void main(String[] args) throws InterruptedException, MalformedURLException {
-        System.out.println("Starting Simple example...");
+        System.out.println("Starting com.lightstep.tracer.jre.example.Simple example...");
```
Comment (danielamiao) [inline examples/src/main/java/com/lightstep/tracer/jre/example/Simple.java]: I feel like there is a misuse of search and replace here....

File: examples/src/main/java/com/lightstep/tracer/jre/example/Simple.java [hunk 2]
```diff
@@ -9,19 +16,13 @@
...
 
         Options options = new Options.OptionsBuilder()
                 .withAccessToken("{your_access_token}")
-                .withComponentName("JRE Simple")
+                .withComponentName("JRE com.lightstep.tracer.jre.example.Simple")
```
Comment (danielamiao) [inline examples/src/main/java/com/lightstep/tracer/jre/example/Simple.java]: same here

File: examples/src/main/java/com/lightstep/tracer/jre/example/Simple.java (hunk 1 above)
Comment (frenchfrywpepper) [inline examples/src/main/java/com/lightstep/tracer/jre/example/Simple.java]: fixed

File: examples/src/main/java/com/lightstep/tracer/jre/example/Simple.java (hunk 2 above)
Comment (frenchfrywpepper) [inline examples/src/main/java/com/lightstep/tracer/jre/example/Simple.java]: fixed

**Pull request merged**